geoprocessing/src/modules/shapefiles/shapefiles.service.ts

```diff
@@ -1,35 +1,17 @@
 import { Injectable, Logger } from '@nestjs/common';
 
-import { createReadStream, rmdirSync, unlinkSync } from 'fs';
-import { Extract } from 'unzipper';
+import { FileService } from '../../utils/file.utils';
+
 const mapshaper = require('mapshaper');
 
 @Injectable()
-export class ShapeFileService {
-  constructor(private readonly logger: Logger) {
-    this.logger.setContext(ShapeFileService.name);
-  }
-
-  private unzipShapefile(fileInfo: Express.Multer.File): Promise<string> {
-    return new Promise((resolve, reject) => {
-      createReadStream(fileInfo.path)
-        .pipe(
-          Extract({
-            path: `${fileInfo.destination}/${fileInfo.filename.replace(
-              '.zip',
-              '',
-            )}`,
-          }),
-        )
-        .on('close', () =>
-          resolve(`${fileInfo.filename} extracted successfully`),
-        )
-        .on('error', (error: Error) =>
-          reject(
-            new Error(`${fileInfo.filename} could not be extracted: ` + error),
-          ),
-        );
-    });
+export class ShapefileService {
+  constructor(
+    private readonly logger: Logger,
+    private fileService: FileService,
+  ) {
+    this.logger.setContext(ShapefileService.name);
```

Comment:
As the NestJS injects the logger as a singleton (in `PlanningUnitsModule` given as a `ClassProvider`) – then it changes the `context` of the logger in all services of the module.

Reply:
@Dyostiq Totally right, I just decided to delete contextsetting and let the controller override it, 1105fe35e9b4ce2eeb8265172f46496f055d8978

Let me know yout thoughts, pls 😃

Reply:
A controller overwriting a context is a bit surprising and very implicit. It will get worse with multiple controllers in a module. Honestly, I would stay with `private readonly logger = new Logger(this.constructor.name)` in every class. [Muting loggers is then still possible, and NestJS handles it as soon as we don't use static methods]. With a logger as a singleton, we should not set the `context` manually, and provide if with every log. If you insist injecting it, the solution is to change the provider to provide a constructor or a factory for the logger, not the instance.

Reply:
@Dyostiq 01859c0bba0333574d75839c6849ff5a3dd27a4b